### PR TITLE
Add YAML formatting standardisation to formatting & linting scripts

### PR DESCRIPTION
**Pull request by strickvl**
I added `yamlfix` to our formatting / linting scripts which applies a standard formatting for our YAML files used in the CI. (They were getting a bit unmanageable + everyone used a different format, so it's worth going with a simple opinionated standard).

We don't have any `yaml` files inside our `src` directory and there's a bug in `yamlfix` when you try to exclude folders, so I haven't included that `src` folder in the scripts.

(This PR also includes fixes following the application of the updated scripts.)

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: .github/actions/setup_environment/action.yml [hunk 1]
```diff
@@ -97,38 +93,36 @@ runs:
...
+      if: steps.cache-virtualenv.outputs.cache-hit != 'true' && steps.custom-cache-pip.outputs.cache-hit
+        != 'true'
       shell: bash
       run: |
         scripts/install-zenml-dev.sh --integrations ${{ inputs.install_integrations }}
 
       # if using a cached virtualenv, just refresh the ZenML installation 
     - name: Refresh ZenML installation
-      if: steps.cache-virtualenv.outputs.cache-hit == 'true' || steps.custom-cache-pip.outputs.cache-hit == 'true'
+      if: steps.cache-virtualenv.outputs.cache-hit == 'true' || steps.custom-cache-pip.outputs.cache-hit
```
Comment (safoinme) [inline .github/actions/setup_environment/action.yml:118]: does GitHub detect this back-to-line without an issue?

File: .github/workflows/publish_api_docs.yml [hunk 2]
```diff
@@ -43,12 +38,10 @@ jobs:
...
           install_integrations: 'no'
-
       - name: Setup git user
         run: |
           git config --global user.name 'Code Monkey'
           git config --global user.email 'codemonkey@zenml.io'
-
       - name: Runs docs generation
-        run: |
+        run: |-
```
Comment (safoinme) [inline .github/workflows/publish_api_docs.yml:46]: what does the `-` do?

**Review by safoinme — APPROVED** [review]

File: .github/workflows/publish_api_docs.yml (hunk 2 above)
Comment (strickvl) [reply .github/workflows/publish_api_docs.yml:46]: @coderabbitai what do you think?

File: .github/actions/setup_environment/action.yml (hunk 1 above)
Comment (strickvl) [reply .github/actions/setup_environment/action.yml:118]: @coderabbitai wdyt?

**Pull request merged**